R/fetch_description.R

```diff
@@ -168,3 +168,111 @@ fetch_description <-
 
     return(output)
   }
+
+
+#'Download and save a description of a survey in Qualtrics Survey Format (QSF)
+#'
+#'@param surveyID String. Unique ID for the survey you want to download.
+#'  Returned as "id" by the [all_surveys] function.
+#'@param formatting String. Adjustments to the downloaded QSF JSON to be either
+#'  more human-readable or more compact.  Options are "none" (default, no
+#'  adjustment), "pretty" (more human-readable, using
+#'  \code{\link[jsonlite]{prettify}}), or "compact" (more compact, using
+#'  \code{\link[jsonlite]{minify}})
+#'@param file String. Desired filename, including path if desired.
+#'  Typical filename extension is .qsf, though not required (and will not be
+#'  added if absent).  Argument ignored if \code{save == FALSE}.
+#'@param save Boolean. Should downloaded QSF be saved to a file (\code{TRUE},
+#'  default), or returned as output (\code{FALSE})?  Even when \code{TRUE},
+#'  downloaded JSON is returned invisibly.
+#'
+#'
+#'@template retry-advice
+#'@return A JSON object equivalent of a Qualtrics QSF, invisibly unless
+#'  \code{save == FALSE}
+#'
+#'@importFrom jsonlite prettify
+#'@importFrom jsonlite minify
+#'@importFrom rlang arg_match
+#'@importFrom stringr str_remove_all
+#'
+#'
+#'@export
+#'
+write_qsf <-
+  function(surveyID,
+           formatting = c("none", "pretty", "compact"),
+           file = NULL,
+           save = TRUE
```

Comment:
I think we've got a little bit of [interacting arguments](https://design.tidyverse.org/args-independence.html) here between `file` and `save`. I think the best option is to make `file = NULL` do the same thing as the current `save = FALSE`. That means the default behavior (with `file = NULL`) would be to return the JSON content but not save it, but if someone provides a `file` then it will save it.